client/opentracing/src/main/java/org/hawkular/apm/client/opentracing/EnvironmentAwareTraceRecorder.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2015-2016 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.apm.client.opentracing;
+
+import org.hawkular.apm.api.model.trace.Trace;
+import org.hawkular.apm.client.api.recorder.TraceRecorder;
+
+/**
+ * @author Juraci Paixão Kröhling
+ */
+public class EnvironmentAwareTraceRecorder implements TraceRecorder {
+    private TraceRecorder backingTraceRecorder;
+    private DeploymentMetaData deploymentMetaData;
+
+    public EnvironmentAwareTraceRecorder(TraceRecorder backingTraceRecorder, DeploymentMetaData deploymentMetaData) {
+        this.deploymentMetaData = deploymentMetaData;
+        this.backingTraceRecorder = backingTraceRecorder;
+    }
+
+    public void report(Trace trace) {
+        backingTraceRecorder.report(trace);
```

Comment:
Rather than checking 'instanceof' EnvAwareTR in TraceContext, could you add the service name and build stamp here? prior to reporting the trace to the backing trace recorder?